formbar/form.py

```diff
@@ -841,66 +842,74 @@ def _get_value(self, value, default, expand):
-            # options are SQLAlchemy based options the variable can be
-            # used to access a attribute of the item. E.g. %id will
-            # access the id of the current option item. For user defined
-            # options "%" can be used to iterate over the user defined
-            # options. In this case %attr will access a given attribte
-            # in the option. A bare "%" will give the value of the
-            # option.
-            if x.startswith("%"):
-                key = x.strip("%")
-                value = "*%s" % (key or "value")
-            # @ marks the item of the current fields form item.
-            elif x.startswith("@"):
-                key = x.strip("@")
-                value = getattr(self._form._item, key)
-            # $ special attributes of the current form.
-            elif x.startswith("$"):
-                tmpitem = None
-                value = None
-                tokens = x.split(".")
-                if len(tokens) > 1:
-                    key = tokens[0].strip("$")
-                    attribute = ".".join(tokens[1:])
-                    # FIXME: This is a bad assumption that there is a
-                    # user within a request. (ti) <2014-07-09 11:18>
-                    if key == "user":
-                        tmpitem = self._form._request.user
-                else:
-                    key = tokens[0].strip("$")
-                    value = item_values.get(key) or ''
-                if tmpitem and not value:
-                    value = getattr(tmpitem, attribute)
-                    if hasattr(value, '__call__'):
-                        value = value()
-            else:
-                value = None
+            key = tokens[0].strip("$")
+            value = substitution_values.get(key) or ''
+        if tmpitem and not value:
+            value = getattr(tmpitem, attribute)
+            if hasattr(value, '__call__'):
+                value = value()
+        return value
 
-            if value is not None:
-                if isinstance(value, list):
-                    value = "[%s]" % ",".join("'%s'"
-                                              % unicode(v) for v in value)
-                    expr_str = expr_str.replace(x, value)
-                elif isinstance(value, basestring) and value.startswith("$"):
-                    expr_str = expr_str.replace(x, "%s" % unicode(value))
-                else:
-                    expr_str = expr_str.replace(x, "'%s'" % unicode(value))
-        return Rule(expr_str.replace("*", "$"))
+    def parse_formitem(self, token):
+        return getattr(self._form._item, token.strip("@"))
+
+    def parse_optionfield(self, token):
+        return "*%s" % (token.strip("%") or "value")
+
+    def determine_values(self):
+        if not self._form._item:
+            return {}
+        values = self._form._item.get_values()
+        values.update(self._form.merged_data)
```

Comment:
I think this one is actually not needed. merged_data is build on form init and already includes the values from get_values